ca/server.go

```diff
@@ -371,15 +393,19 @@ func (s *Server) Run(ctx context.Context) error {
 		}).WithError(err).Errorf("error attempting to reconcile certificates")
 	}
 
-	// Watch for changes.
+	// Watch for changes in the cluster membership state.
 	for {
 		select {
 		case event := <-updates:
 			switch v := event.(type) {
 			case state.EventCreateNode:
 				s.evaluateAndSignNodeCert(ctx, v.Node)
 			case state.EventUpdateNode:
-				s.evaluateAndSignNodeCert(ctx, v.Node)
+				// If this certificate is already at a final state
+				// no need to evaluate and sign it.
+				if !isFinalState(v.Node.Certificate.Status) {
```

Comment:
The problem with Update is potential for loops. If create executes `s.evaluateAndSignNode` unnecessarily, that isn't a big issue.